dags/analyzer_helper/tests/integration/test_extract_raw_info.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+import unittest
+from analyzer_helper.discourse.extract_raw_info import ExtractRawInfo
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
+from github.neo4j_storage.neo4j_connection import Neo4jConnection
+
+class TestExtractRawInfo(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.neo4jConnection = Neo4jConnection()
+        cls.client = MongoSingleton.get_instance().client
+        cls.driver = cls.neo4jConnection.connect_neo4j()
+        cls.forum_endpoint = "http://test_forum" 
+        cls.platform_id = "platform_db"       
+        cls.platform_db = cls.client[cls.platform_id]
+        cls.extractor = ExtractRawInfo(cls.forum_endpoint, cls.platform_id)
+        cls.rawmemberactivities_collection = cls.platform_db["rawmemberactivities"]
+
+        with cls.driver.session() as session:
+            session.run("MATCH (n) DETACH DELETE n")
+
+        with cls.driver.session() as session:
+            session.run("""
+                CREATE (f:DiscourseForum {endpoint: $endpoint, uuid: 'forum-uuid'}),
+                       (u1:DiscourseUser {id: 'user1', name: 'User One'}),
+                       (u2:DiscourseUser {id: 'user2', name: 'User Two'}),
+                       (p1:DiscoursePost {id: '1', content: 'Post 1', createdAt: '2023-01-01T00:00:00Z', topicId: 'topic-uuid'}),
+                       (p2:DiscoursePost {id: '2', content: 'Post 2', createdAt: '2023-01-02T00:00:00Z', topicId: 'topic-uuid'}),
+                       (t:DiscourseTopic {id: 'topic-uuid', forumUuid: 'forum-uuid'}),
+                       (c:DiscourseCategory {id: 'category1', name: 'Category 1'}),
+                       (p1)<-[:HAS_POST]-(t),
+                       (p2)<-[:HAS_POST]-(t),
+                       (p1)<-[:POSTED]-(u1),
+                       (p2)<-[:POSTED]-(u2),
+                       (p1)<-[:LIKED]-(u2),
+                       (p2)<-[:REPLY_TO]-(p1),
+                       (c)-[:HAS_TOPIC]->(t)
+                """, {'endpoint': cls.forum_endpoint})
+
+    @classmethod
+    def tearDownClass(cls):
+        with cls.driver.session() as session:
+            session.run("MATCH (n) DETACH DELETE n")
+        cls.extractor.close()
+        cls.driver.close()
+
+    def test_fetch_post_details(self):
+        post_details = self.extractor.fetch_post_details()
+        self.assertEqual(len(post_details), 2)
+        author_names = [post['author_name'] for post in post_details]
```

Comment:
We don't want to fetch `author_name` to be extracted from posts. You could do the assertion on post ids.